src/components/Link.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import { timeDifferenceForDate } from '../utils';
+import { Mutation } from 'react-apollo';
+import gql from 'graphql-tag';
+
+import { AUTH_TOKEN } from '../constants';
+
+const VOTE_MUTATION = gql`
+  mutation VoteMutation($linkId: ID!) {
+    vote(linkId: $linkId) {
+      id
+      link {
+        votes {
+          id
+          user {
+            id
+          }
+        }
+      }
+      user {
+        id
+      }
+    }
+  }
+`;
+
+class Link extends Component {
+  //This is an extremely uggly fix. I know.
+  reloadPage = () => {
+    window.location.reload();
```

Comment:
About the issue you're having updating the apollo client side cache, did you try using `refetchQueries` on the mutation? 

More info, in case you haven't checked this part of the docs yet: https://www.apollographql.com/docs/react/essentials/mutations/#updating-the-cache

Reply:
Thank you for leading me to this part of the docs and your suggestion! I will try to learn this part a bit better and see if I can resolve the issue I'm having 😊